### ISSUE TITLE
Pruning fails on empty/singleton supports and skips complete deletion

### ISSUE THREAD

**Issue by sanghack81** [soheunyi/frp#2]
The pruning prefix searches in `frp/rank_and_prune.py` do not cover zero through all remaining edges:

- With `search="sequential"`, zero or one candidate leaves `n_del` unassigned and raises `UnboundLocalError`.
- With `search="binary"`, a removable last edge is retained.
- For larger supports, the sequential search also excludes testing the complete deletion prefix.

Reproduced against `88b4455eb3164418d2513054d53a5d847c6edcf7`.

### Minimal reproduction

This uses an exact loss oracle to isolate pruning from optimizer behavior. For target precision `Theta = I`, the SEM `W = 0, Omega = I` is feasible under every allowed support, so `L(E, I) = 0`.

```python
import numpy as np
from frp.rank_and_prune import RankAndPrune

for search in ["sequential", "binary"]:
    for n_edges in [0, 1]:
        support = np.eye(2)
        support[0, 1] = n_edges
        manager = RankAndPrune(
            np.eye(2), support, "kld", "zero", {},
            edge_penalty=0.1, loss_inc_tol=0.1,
            search=search, use_rank=False, use_loss_cache=False,
        )
        manager.no_penalty_loss_estimator.estimate_loss = (
            lambda support, theo_best=-np.inf: (0.0, np.eye(2))
        )
        try:
            result = manager.rank_and_prune_once()
            remaining = np.count_nonzero(result - np.diag(np.diag(result)))
            print(f"{search}: {n_edges} input edges -> {remaining} remaining")
        except UnboundLocalError:
            print(f"{search}: {n_edges} input edges -> UnboundLocalError")
```

Actual output:

```text
sequential: 0 input edges -> UnboundLocalError
sequential: 1 input edges -> UnboundLocalError
binary: 0 input edges -> 0 remaining
binary: 1 input edges -> 1 remaining
```

Expected: no exception and zero remaining edges in all four cases. Both searches should consider accepted prefix lengths from zero through the entire candidate count, while retaining the existing single-edge fallback if the first prefix is rejected.

This report concerns pruning control flow. The paper's historical benchmarks have not been rerun. The reproduction and proposed fix were prepared and checked with AI assistance.
